EventFilter/HGCalRawToDigi/plugins/HGCalRawToDigi.cc

```diff
@@ -116,6 +130,30 @@ void HGCalRawToDigi::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
   // put information to the event
   iEvent.emplace(digisToken_, std::move(digis));
   iEvent.emplace(econdPacketInfoToken_, std::move(econdPacketInfo));
+  iEvent.emplace(fedPacketInfoToken_, std::move(fedPacketInfo));
+}
+
+//
+uint16_t HGCalRawToDigi::callUnpacker(unsigned fedId,
+                                      const FEDRawData& fed_data,
+                                      const HGCalMappingModuleIndexer& moduleIndexer,
+                                      const HGCalConfiguration& config,
+                                      hgcaldigi::HGCalDigiHost& digis,
+                                      hgcaldigi::HGCalECONDPacketInfoHost& econdPacketInfo) {
+  uint16_t status(0);
+  try {
```

Comment:
can you rely on a failure exit code instead of an exception ?

Reply:
I would have some preference to leave it for the time being as an exception

Reply:
I'd recommend to look into moving away from exception rather sooner than later. Exceptions should be reserved for truly exceptional cases (7.11 in https://cms-sw.github.io/cms_coding_rules.html, https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideEdmExceptionUse#Exception_Handling). Exceptions provide slow control flow, and when used for control flow they make investigations of real exceptions more difficult.

Reply:
ok, point taken, I'm doing a few more checks before releasing the try / catch

Reply:
ok with a better understanding of the situation I could remove the try / catch.
I leave the call to the unpacker as a separate function for now, it may be that additional checks will be needed in the future.